contrib/epee/src/net_parse_helpers.cpp

```diff
@@ -153,7 +153,14 @@ namespace net_utils
     }
     if(result[6].matched)
     {
-      content.port = boost::lexical_cast<uint64_t>(result[6]);
+      const auto port_str = result[6].str();
```

Comment:
Would help to have a port parsing function, then this becomes:
```
if (!parse_port(result[6].str(), content.port)) {
    return false;
}
```

Also removed the logging from the example, as I don't think parsing functions should be in the business of logging. If the caller wants info on what failed, there are plenty of well-worn options, like providing an out param, or an error code.

Reply:
Added a function and removed logging.